npe2/manifest/schema.py

```diff
@@ -376,6 +378,84 @@ def _from_package_or_name(
 
     ValidationError = ValidationError  # for convenience of access
 
+    @classmethod
+    def _from_npe1_plugin(cls, plugin_name: str):
+        from .._from_npe1 import npe1_plugin_manager  # type: ignore
+
+        plugin_manager = npe1_plugin_manager()
+        if plugin_name not in plugin_manager.plugins:
+            # TODO: it would be nice to add some logic to prevent confusion here.
+            # for example... if the plugin name doesn't equal the package name, we
+            # should still be able to find it if the user gives a package name
+            raise ValueError(
+                f"Could not find plugin {plugin_name!r}\n"
+                f"Found {set(plugin_manager.plugins)}"
+            )
+
+        module = plugin_manager.plugins[plugin_name]
+        standard_meta = plugin_manager.get_standard_metadata(plugin_name)
+        package = standard_meta["package"].replace("-", "_")
+
+        commands = []
+        readers = []
+        writers = []
+        for caller in plugin_manager._plugin2hookcallers[module]:
+            for impl in caller.get_hookimpls():
+                if impl.plugin_name != plugin_name:
+                    continue
+                name = impl.specname.replace("napari_", "")
+                id = f"{package}.{name}"
+                py_name = f"{impl.function.__module__}:{impl.function.__qualname__}"
+                title = " ".join(name.split("_")).title()
+                commands.append({"id": id, "python_name": py_name, "title": title})
+
+                if "reader" in id:
+                    # add this to the readers list
+                    readers.append({"command": id, "accepts_directories": True})
+
+                write_layers = []
+                write_directory = False
+                if "write" in id:
+                    # add this to the writers list
+                    if "get_writer" not in id:
+                        layer = id.split("write_")[1]
+                        if layer != "directory":
```

Comment:
`napari_write_directory` is a not a hookspec, so won't `id.split("write_")[1]` != 'directory'` _always_ be true?

(meaning `if write_directory` will never happen)?

Reply:
I got this from looking at this: https://github.com/napari/napari/blob/main/napari/builtins.yaml

Its not anywhere on the old napari plugin engine?  I will remove it.

Reply:
the "directory writer" is basically I name I made up when authoring builtins.yaml.  The corresponding thing in the original plugin engine is the napari_get_writer implementation [here](https://github.com/napari/napari/blob/c01794a3faf11c6c8f5b9833aae55b38a24bd43b/napari/plugins/_builtins.py#L259). It happens to work by creating a directory and saving layers inside of it.  

But the question here is what to do with `napari_get_writer` hooks.  

It's not clear how to translate these automatically. The writer contribution basically binds the things that `napari_get_writer` returns rather than `napari_get_writer` itself.